halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/GlobalApplicationOptions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.core;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Data;
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This is the collection of general, top-level flags that come from the application configuration for halyard.
+ * This class is both @AutoWired'able and can act as a singleton via the getInstance() function. This is so the same
+ * class may be reused between halyard's CLI and daemon.
+ */
+@Data
+@Configuration
+@PropertySource("file:" + GlobalApplicationOptions.CONFIG_PATH)
+public class GlobalApplicationOptions {
+
+  static final String CONFIG_PATH = "/opt/spinnaker/hal.properties";
```

Comment:
Is this config shared between the client and the server? Based on the javadoc string, it appears so.

All of the other components follow a convention for server configuration options that hooks into Spring's parsing and options logic (here's a guide on how that works: https://www.travistomsu.com/2016/12/19/configuring-spinnaker/)

If possible, I'd suggest using the yaml format instead of `.properties` files.

Reply:
The reason this is done this way is because the CLI doesn't use spring and i'd prefer to share the implementation between the two.

Reply:
Also its still Autowire-ed in the halyard-web backend.